openmdao/recorders/tests/test_sqlite_recorder.py

```diff
@@ -384,6 +384,77 @@ def test_driver_records_metadata(self):
         }
         assertDriverMetadataRecorded(self, expected_driver_metadata)
 
+    def test_system_records_no_metadata(self):
+        prob = Problem(model=SellarDerivatives())
+
+        recorder = SqliteRecorder("cases.sql")
+        prob.model.add_recorder(recorder)
+        prob.model.recording_options['record_model_metadata'] = False
+        prob.model.recording_options['record_metadata'] = False
+
+        prob.setup()
+        prob.run_model()
+        prob.cleanup()
+
+        cr = CaseReader("cases.sql")
+        self.assertEqual(len(cr.system_metadata.keys()), 0)
+
+    def test_system_record_model_metadata(self):
+        # first check to see if recorded recursively, which is the default
+        prob = Problem(model=SellarDerivatives())
+        prob.setup()
+
+        recorder = SqliteRecorder("cases.sql")
+        prob.model.add_recorder(recorder)
+
+        prob.run_model()
+        prob.cleanup()
+
+        cr = CaseReader("cases.sql")
+        self.assertEqual(set(cr.system_metadata.keys()),
+                         set(['root', 'px', 'pz', 'd1', 'd2', 'obj_cmp', 'con_cmp1', 'con_cmp2']))
```

Comment:
To be safe on all these tests, maybe you want to check the values too?

Reply:
We could but a lot of the other tests did not. So if we do this one, shouldn't we do the other ones?

Reply:
To be safe, yes.

Reply:
ok